pandas/core/series.py

```diff
@@ -1576,6 +1576,56 @@ def __repr__(self) -> str:
 
         return result
 
+    def _repr_html_(self):
+        # TODO: Full independent HTML generation in SeriesFormatter, rather
+        # than depending on a limited subset of functionality via to_frame().
+
+        if get_option("display.notebook_repr_html"):
+            max_rows = get_option("display.max_rows")
+            min_rows = get_option("display.min_rows")
+            show_dimensions = get_option("display.show_dimensions")
+
+            formatter = fmt.DataFrameFormatter(
+                self.to_frame(),
+                columns=None,
+                col_space=None,
+                na_rep="NaN",
+                formatters=None,
+                float_format=None,
+                sparsify=None,
+                justify=None,
+                index_names=True,
+                header=False,
+                index=True,
+                bold_rows=True,
+                escape=True,
+                max_rows=max_rows,
+                min_rows=min_rows,
+                show_dimensions=False,  # We do this later for a series.
+                decimal=".",
+                table_id=None,
+                render_links=False,
+            )
+            html = formatter.to_html(notebook=True).split("\n")
+
+            # Find out where the column ends - we will insert footer information here.
+            tbl_end = [
```

Comment:
@big-o I think would be better on this PR if you can do this *inside* of DataFrameFormatter in its construction itself, rather than 'finding' then end later. is this possible?

Reply:
Do you mean adding an extra `footer` parameter to either `DataFrameFormatter.__init__` or `DataFrameFormatter.to_html`? Since the `DataFrameFormatter` isn't limited to single column, data, what should be footer look like for frames with multiple columns? Should there even be one? Should an Exception be raised? Would appreciate some clear guidance on what is expected before I start on another rewrite, as I only have limited time to spend on this.